### ISSUE TITLE
ENH: config.patch context manager

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#5155]
CLI has `datalad -c var=val ` which makes `var=val` in effect only for the duration of that `datalad` call.
In Python API AFAIK we do not have such a convenience, and only `patch_config` helper for tests.

I think a good design pattern could be to just move that `patch_config` to become a context manager within `Config` itself, so we could do

```python
from datalad import cfg
with cfg.patch({'var': val, ...}):
    ...
```

That would also avoid "touching" protected `cfg._merged_store` within `datalad.tests.utils` thus keeping access to protected attributes within Config

WDYT @datalad/developers ?

**Comment by mih** [comment]
Makes sense. It may be necessary to actually patch `.overrides` and trigger a reload going in an out of the context. Otherwise the patch will not survive a reload that happens in the context.

**Comment by mih** [comment]
The is an (alternative) proposal that would likely make internal use of the feature proposed here

- https://github.com/datalad/datalad/issues/5587

**Comment by yarikoptic** [comment]
FTR, we have `tests.utils.patch_config` https://github.com/datalad/datalad/blob/HEAD/datalad/tests/utils.py#L2071 which pretty much provides this functionality but better be not used outside of tests.

**Comment by mih** [comment]
https://github.com/datalad/datalad-next/pull/760 now has a context manager that can do this. It posts overrides to the Git-aware ENV, hence affects subprocesses also. Looks like this:

```py
    with manager.overrides(cfg_overrides):
        ...
```

where `cfg_overrides` is a `dict` with overrides and `manager` is the incoming config maanager based on https://github.com/datalad/datasalad/pull/47
